pkg/controller/sesv2/configurationset/setup.go

```diff
@@ -0,0 +1,345 @@
+			managed.WithPollInterval(o.PollInterval),
+			managed.WithLogger(o.Logger.WithValues("controller", name)),
+			managed.WithRecorder(event.NewAPIRecorder(mgr.GetEventRecorderFor(name)))))
+}
+
+type hooks struct {
+	client sesv2iface.SESV2API
+	kube   client.Client
+}
+
+func isUpToDate(cr *svcapitypes.ConfigurationSet, resp *svcsdk.GetConfigurationSetOutput) (bool, error) {
+	if meta.WasDeleted(cr) {
+		return true, nil // There is no need to check for updates when we want to delete.
+	}
+
+	if !isUpToDateDeliveryOptions(cr, resp) {
+		return false, nil
+	}
+
+	if !isUpToDateReputationOptions(cr, resp) {
+		return false, nil
+	}
+
+	if !isUpToDateSendingOptions(cr, resp) {
+		return false, nil
+	}
+
+	if !isUpToDateSuppressionOptions(cr, resp) {
+		return false, nil
+	}
+
+	if !isUpToDateTrackingOptions(cr, resp) {
+		return false, nil
+	}
+
+	return svcutils.AreTagsUpToDate(cr.Spec.ForProvider.Tags, resp.Tags)
+}
+
+// isUpToDateDeliveryOptions checks if DeliveryOptions Object are up to date
+func isUpToDateDeliveryOptions(cr *svcapitypes.ConfigurationSet, resp *svcsdk.GetConfigurationSetOutput) bool {
+	// User may remove DeliveryOptions spec and expect an update
+	if cr.Spec.ForProvider.DeliveryOptions == nil && resp.DeliveryOptions != nil {
+		cr.Spec.ForProvider.DeliveryOptions = &svcapitypes.DeliveryOptions{}
+		return false
+	}
+
+	if cr.Spec.ForProvider.DeliveryOptions != nil && resp.DeliveryOptions != nil {
+		if awsclient.StringValue(cr.Spec.ForProvider.DeliveryOptions.SendingPoolName) != awsclient.StringValue(resp.DeliveryOptions.SendingPoolName) {
+			return false
+		}
+		if awsclient.StringValue(cr.Spec.ForProvider.DeliveryOptions.TLSPolicy) != awsclient.StringValue(resp.DeliveryOptions.TlsPolicy) {
+			return false
+		}
+	}
+	return true
+}
+
+// isUpToDateReputationOptions checks if ReputationOptions Object are up to date
+func isUpToDateReputationOptions(cr *svcapitypes.ConfigurationSet, resp *svcsdk.GetConfigurationSetOutput) bool {
+	// User may remove ReputationOptions spec and expect an update
+	if cr.Spec.ForProvider.ReputationOptions == nil && resp.ReputationOptions != nil {
+		cr.Spec.ForProvider.ReputationOptions = &svcapitypes.ReputationOptions{}
+		return false
+	}
+
+	if cr.Spec.ForProvider.ReputationOptions != nil && resp.ReputationOptions != nil {
+		if awsclient.BoolValue(cr.Spec.ForProvider.ReputationOptions.ReputationMetricsEnabled) != awsclient.BoolValue(resp.ReputationOptions.ReputationMetricsEnabled) {
+			return false
+		}
+	}
+	return true
+}
+
+// isUpToDateTrackingOptions checks if TrackingOptions Object are up to date
+func isUpToDateTrackingOptions(cr *svcapitypes.ConfigurationSet, resp *svcsdk.GetConfigurationSetOutput) bool {
+	// User may remove TrackingOptions spec and expect an update
+	if cr.Spec.ForProvider.TrackingOptions == nil && resp.TrackingOptions != nil {
+		cr.Spec.ForProvider.TrackingOptions = &svcapitypes.TrackingOptions{}
```

Comment:
`cr` should not be changed during `isUpTodate`.

Reply:
have fixed this issue